wger.py

```diff
@@ -127,186 +161,188 @@ def create_exercise(self):
         """The function delete an exercise from a specific training day"""
-        exercises=[]
-        for i in range(20,420,20):
-            exercises.extend(self.get_info(f'exercise/?limit=20&offset={i}')['results'])
+        exercises=self.get_info('exercise/',420)['results']
         workouts=self.get_info('workout/')['results']
         days=self.get_info('day/')['results']
-        sets=self.get_info('set/')['results']
-        settings=self.get_info('setting/')['results']
-        ex_day=[]
-
-        id_exercise = ""
-        id_workout=""
-        id_day=""
+        sets=self.get_info('set/',41)['results']
+        settings=self.get_info('setting/',41)['results']
+        day_ex=[]
         id=""
 
-        for i in workouts:
-            if i['name']==workout:
-                id_workout=i['id']
-        if id_workout == "":
-            return "Workout not found"
+        id_workout=self.match(workouts,workout, 'name', 'id')
+        if id_workout=="":
+            return (False,"Workout not found")
 
-        for i in days:
-            if i['description'] == day:
-                id_day = i['id']
-        if id_day == "":
-            return "Training not found"
+        if day==None:
+            return self.delete_workout(workout)
+        id_day=self.match(days, day, 'description', 'id')
+        if id_day=="":
+            return (False,"Day not found")
 
-        for i in exercises:
-            if i['name']==exercise:
-                id_exercise=i['id']
-        if id_exercise== "":
-            return "Exercise does not exists"
+        if exercise==None:
+            return self.delete_day(workout,day)
+        id_exercise=self.match(exercises,exercise, 'name', 'id')
+        if id_exercise=="":
+            return (False,"Exercise not found")
 
         for i in sets:
             if i['exerciseday']==id_day:
-                ex_day.append(i['id'])
+                day_ex.append(i.get('id'))
         for i in settings:
-            if i['set'] in ex_day and i['exercise']==id_exercise:
-                id=i['id']
+            if i['exercise']==id_exercise and i['set'] in day_ex:
+                id=i.get('set')
 
-        self.delete_info(f'setting/{id}/')
-        return f"Exercise {exercise} was deleted successfully!"
+        self.delete_info(f'set/{id}')
+        return any(i['set'] for i in self.get_info('set/').get('results')if i['sets']==id)==False
+    
+    def delete_day(self,workout,day):
+        """This function delete a training day"""
+        id_workout=self.match(self.get_info('workout/').get('results'),workout,'name','id')
+        days=[i for i in self.get_info('day/').get('results') if i['training']==id_workout]
+        id_day=self.match(days,day,'description','id')
+        if id_day=="":
+            return (False,"Day not found")
+        self.delete_info(f"day/{id_day}")
+        return any([i['id'] for i in self.get_info('day/').get('results') if i['id']==id_day])==False
```

Comment:
You should combine this logic with the response of the 'delete_info'.
This should be done in order to make sure the delete request returned an acceptable response code.